### ISSUE TITLE
Can't update AWS account identifiers.  Also, SM should remove any hyphens in identifiers.

### ISSUE THREAD

**Issue by bbirkinbine** [Netflix/security_monkey#783]
I kept getting the following error even though I checked my IAM role settings and was able to manually sts.assume_role() via boto3 from the ec2 instance.

ClientError: An error occurred (AccessDenied) when calling the AssumeRole operation: Not authorized to perform sts:AssumeRole

"""
2017-08-07 01:50:33,284 ERROR: Job "run_change_reporter (trigger: interval[1:00:00], next run at: 2017-08-07 02:50:31.875327)" raised an exception [in build/bdist.linux-x86_64/egg/apscheduler/scheduler.py:520]
Traceback (most recent call last):
  File "build/bdist.linux-x86_64/egg/apscheduler/scheduler.py", line 512, in _run_job
    retval = job.func(*job.args, **job.kwargs)
  File "/usr/local/src/security_monkey/security_monkey/scheduler.py", line 32, in run_change_reporter
    reporter.run(account, interval)
  File "/usr/local/src/security_monkey/security_monkey/reporter.py", line 56, in run
    (items, exception_map) = monitor.watcher.slurp()
  File "/usr/local/src/security_monkey/security_monkey/cloudaux_watcher.py", line 77, in slurp
    regions=self._get_regions(), conn_type='dict')
  File "/usr/local/src/security_monkey/security_monkey/cloudaux_watcher.py", line 38, in _get_regions
    _, regions = get_regions(account, self.service_name)
  File "/usr/local/src/security_monkey/security_monkey/decorators.py", line 168, in get_regions
    role = sts.assume_role(RoleArn='arn:aws:iam::' + account.identifier + ':role/' + role_name, RoleSessionName='secmonkey')
  File "/usr/local/src/security_monkey/venv/lib/python2.7/site-packages/botocore-1.5.92-py2.7.egg/botocore/client.py", line 310, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/usr/local/src/security_monkey/venv/lib/python2.7/site-packages/botocore-1.5.92-py2.7.egg/botocore/client.py", line 599, in _make_api_call
    raise error_class(parsed_response, operation_name)
ClientError: An error occurred (AccessDenied) when calling the AssumeRole operation: Not authorized to perform sts:AssumeRole
"""

decorators.py near line 170 has the following line
role = sts.assume_role(RoleArn='arn:aws:iam::' + account.identifier + ':role/' + role_name, RoleSessionName='secmonkey')

account.identifier in the database has the account with hyphens 'NNNN-NNNN-NNNN' instead of 'NNNNNNNNNNNN' so the sts.assume_role() will not properly work.

Not sure if something has changed recently where the DB insert for account.identifier includes the account id with hyphens. 

I believe I added the AWS account through the WebUI and when I put in the accountid, it seems to reformat it to include the hyphens no matter what I try.

If I do a DB update to remove the hyphens, the sts:AssumeRole error goes away

`
-[ RECORD 39 ]--+-----------------------------------------------------------------------------------------------
id              | 1
active          | t
third_party     | f
name            | redacted
notes           | 
account_type_id | 1
identifier      | NNNN-NNNN-NNNN

update account set identifier = 'NNNNNNNNNNN' where identifier = 'NNNN-NNNN-NNNN'

-[ RECORD 39 ]--+-----------------------------------------------------------------------------------------------
id              | 1
active          | t
third_party     | f
name            | redacted
notes           | 
account_type_id | 1
identifier      | NNNNNNNNNNNN
`

**Comment by monkeysecurity** [comment]
AccountManager.update() doesn't currently allow you to update account identifiers.

Also, SM should strip hyphens in AWS account identifiers before saving to the DB.

**Comment by bbirkinbine** [comment]
Thanks for the explanation, was able to fix manually in the DB for my case.  I probably copy/pasted from Amazon Web UI the account identifier (which includes hyphens by default).

**Comment by Alofoxx** [comment]
you should also add trimming to remove whitespaces around the account identifier input when saving to db on ui or cli.
The identifier for my aws account in the Account table had a leading space " 000000000000"

**Comment by monkeysecurity** [comment]
Solution allows each account type to decide how to sanitize their identifiers.

For AWS:

```python
def sanitize_account_identifier(self, identifier):
    """AWS identifer sanitization will strip and remove any hyphens.

    Returns:
        stripped identifier with hyphens removed
    """
    return identifier.replace('-', '').strip()
```

**Comment by monkeysecurity** [comment]
Fixed in #784
